### PR TITLE
Changelog Tool Overhaul

### PR DESCRIPTION
**Pull request by N3X15**
<s>**DNM: REQUIRES CI TO BE UPDATED PROBABLY**</s> Done, apparently

I think Pomf or somebody requested this.

## Changes
* Jinja2 templates, which fixes (P)XSS and makes it a LOT easier for other communities to change the style.  Wasn't causing security problems per se, and the threat is very low, but we were getting shit breaking because of various HTML special characters not being escaped.
* PyYAML 5.1, which fixes security vulns (This was already done at some point but now it's official I guess)
* MANDATORY Python 3.6+ support, because Python 2 is EOL.
* Slightly cleaner code
* YAML Schemas for changelogs.  There for the supernerds.
* Changelog will halt processing if an incorrect change prefix is detected.
* An actual fucking README.
[bugfix][system]

## Changelog
*Included as file*
```yaml
author: N3X15
changes:
  - tweak: Changelog tool reworked to be slightly less shit, use python 3.6 and proper logging.
  - tweak: Changelog tool now uses Jinja2 to render actual, whole-assed templates.
  - tweak: Changelog templates coalesced. Still use XHTML 4.01 Transitional because BYOND is shit.
  - bugfix: Fixed some minor low-threat XSS vulnerabilities in changelogs that caused occasional formatting issues.
  - rscadd: Validation schemas added for changelogs, for the poor tortured souls that desire them for whatever godforsaken reason.
```

### CODE REVIEW

File: tools/changelog/ss13_genchangelog.py [hunk 1]
```diff
@@ -1,12 +1,14 @@
-#!/usr/bin/env python2
```
Comment (PJB3005) [inline tools/changelog/ss13_genchangelog.py:1]: You should still keep this shebang, change it to python3 instead of removing it.

**Comment by PJB3005** [comment]
I updated WebMoMMI to actually directly execute the .py file instead of running `python2` with it, and installed the python 3 dependencies, so everything should be good on my end to have this keep working when merged.

File: tools/changelog/ss13_genchangelog.py (hunk 1 above)
Comment (N3X15) [reply tools/changelog/ss13_genchangelog.py]: Many distros no longer ship `env`, so I'm looking for a better alternative shebang.

**Review by PJB3005 — APPROVED** [review]

**Comment by Probe1** [comment]
Should the DNM tag be removed too @N3X15 ?

**Comment by N3X15** [comment]
@Probe1 Yes

**Pull request merged**